detectorSegmentations/src/FCCSWHCalPhiTheta_k4geo.cpp

```diff
@@ -0,0 +1,811 @@
+  registerIdentifier("identifier_layer", "Cell ID identifier for layer", m_layerID, "layer");
+}
+
+FCCSWHCalPhiTheta_k4geo::FCCSWHCalPhiTheta_k4geo(const BitFieldCoder* decoder) : GridTheta_k4geo(decoder) {
+  // define type and description
+  _type = "FCCSWHCalPhiTheta_k4geo";
+  _description = "Phi-theta segmentation in the global coordinates";
+
+  // register all necessary parameters (additional to those registered in GridTheta_k4geo)
+  registerParameter("phi_bins", "Number of bins phi", m_phiBins, 1);
+  registerParameter("offset_phi", "Angular offset in phi", m_offsetPhi, 0., SegmentationParameter::AngleUnit, true);
+  registerParameter("detLayout", "The detector layout (0 = Barrel; 1 = Endcap)", m_detLayout, -1);
+  registerParameter("offset_z", "Offset in z-axis of the layer center", m_offsetZ, std::vector<double>());
+  registerParameter("width_z", "Width in z of the layer", m_widthZ, std::vector<double>());
+  registerParameter("offset_r", "Offset in radius of the layer (Rmin)", m_offsetR, std::vector<double>());
+  registerParameter("numLayers", "Number of layers", m_numLayers, std::vector<int>());
+  registerParameter("dRlayer", "dR of the layer", m_dRlayer, std::vector<double>());
+  registerIdentifier("identifier_phi", "Cell ID identifier for phi", m_phiID, "phi");
+  registerIdentifier("identifier_layer", "Cell ID identifier for layer", m_layerID, "layer");
+}
+
+
+/** /// determine the global position based on the cell ID
+Vector3D FCCSWHCalPhiTheta_k4geo::position(const CellID& cID) const {
+  uint layer = _decoder->get(cID,m_layerID);
+  double radius = 1.0;
+
+  if(m_radii.empty()) defineCellsInRZplan();
+  if(!m_radii.empty()) radius = m_radii[layer];
+
+  return positionFromRThetaPhi(radius, theta(cID), phi(cID));
+}
+**/
+
+/// determine the global position based on the cell ID
+/// returns the geometric center of the cell
+Vector3D FCCSWHCalPhiTheta_k4geo::position(const CellID& cID) const {
+  uint layer = _decoder->get(cID,m_layerID);
+  int thetaID = _decoder->get(cID,m_thetaID);
+  double zpos = 0.;
+  double radius = 1.0;
+
+  if(m_radii.empty()) defineCellsInRZplan();
+  if(!m_radii.empty()) radius = m_radii[layer];
+  if(!m_cellEdges.empty()) zpos = m_cellEdges[layer][thetaID].first + (m_cellEdges[layer][thetaID].second - m_cellEdges[layer][thetaID].first) * 0.5;
+
+  auto pos = positionFromRThetaPhi(radius, theta(cID), phi(cID));
+
+  // return the position with corrected z coordinate to match to the geometric center
+  return Vector3D(pos.x(),pos.y(),zpos);
+}
+
+
+void FCCSWHCalPhiTheta_k4geo::defineCellsInRZplan() const {
+  if(m_radii.empty())
+  {
+    // check if all necessary variables are available
+    if(m_detLayout==-1 || m_offsetZ.empty() || m_widthZ.empty() || m_offsetR.empty() || m_numLayers.empty() || m_dRlayer.empty())
+    {
+       dd4hep::printout(dd4hep::ERROR, "FCCSWHCalPhiRow_k4geo","Please check the readout description in the XML file!\n%s",
```

Comment:
should be PhiTheta... also in following if (..) blocks

Reply:
thanks for spotting that, I have fixed it